src/iperf_error.c

```diff
@@ -94,10 +94,6 @@ iperf_errexit(struct iperf_test *test, const char *format, ...)
     struct tm *ltm = NULL;
     char *ct = NULL;
 
-    if (pthread_mutex_lock(&(test->print_mutex)) != 0) {
```

Comment:
OK you want to move the mutex operation below the (potential) call to `iperf_json_finish()`. This is good.